src/components/lightbox/lightbox.js

```diff
@@ -0,0 +1,91 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import eventedShowHideState from '../../globals/js/mixins/evented-show-hide-state';
+import eventMatches from '../../globals/js/misc/event-matches';
+import on from '../../globals/js/misc/on';
+// import eventMatches from '../../globals/js/misc/event-matches';
+
+class Lightbox extends mixin(createComponent, initComponentBySearch) {
+  constructor(element, options) {
+    super(element, options);
+    this.activeIndex = this.element.dataset.lightboxIndex;
+    this.totalSlides =
+      [...this.element.querySelectorAll(this.options.selectorLightboxItem)].length - 1;
+
+    this.updateSlide();
+    this.updateFilmstrip();
+
+    this.element.addEventListener('click', evt => {
+      this.handleClick(evt);
+    });
+
+    this.element.parentNode.addEventListener('modal-beingshown', evt => {
```

Comment:
Is the modal integration critical? If so, we should throw if `this.element.parentNode` is missing. If not, we should check if `this.element.parentNode` is there to see if we should hook an event listener there. Please see: https://github.com/carbon-design-system/carbon-components/pull/12

Reply:
Another thought - The lightbox demo gave me an impression that the component is different flavor of pop-up, instead of a component composing modal and carousel. If it’s correct, we have another approach - Inheriting modal.